perception/sub8_perception/nodes/follow_orange_pipes.py

```diff
@@ -1,318 +1,269 @@
 #!/usr/bin/env python
 import cv2
 import numpy as np
-import sys
 import rospy
-import rospkg
-import sub8_ros_tools
-import tf
-import os
+from tf.transformations import quaternion_from_euler
 from std_msgs.msg import Header
 from std_srvs.srv import SetBool, SetBoolResponse
-from sub8_msgs.srv import VisionRequestResponse, VisionRequest
+from sub8_msgs.srv import VisionRequestResponse, VisionRequest, VisionRequest2D, VisionRequest2DResponse, VisionRequest, VisionRequestResponse
 from geometry_msgs.msg import PoseStamped
-from sub8_vision_tools import MarkerOccGrid, rviz, machine_learning
-from sub8_ros_tools import numpy_quat_pair_to_pose
+from sub8_ros_tools import numpy_quat_pair_to_pose, numpy_to_point, numpy_matrix_to_quaternion, numpy_to_quaternion
+from sub8_ros_tools import Image_Publisher, Image_Subscriber
 from image_geometry import PinholeCameraModel
-
-SEARCH_DEPTH = .65  # m
-
-rospack = rospkg.RosPack()
-
-class MarkerFinder():
+from visualization_msgs.msg import Marker
+
+__author__ = "Kevin Allen"
+
+class PathMarkerFinder():
+    """
+    Node which finds orange path markers in image frame, 
+    estimates the 2d and 3d position/orientation of marker
+    and returns this estimate when service is called.
+    
+    Unit tests for this node is in test_path_marker.py
+    
+    Finding the marker works as follows:
+    * blur image
+    * threshold image mostly for highly saturated, orange/yellow/red objects
+    * run canny edge detection on thresholded image
+    * find contours on edge frame
+    * filter contours to find those that may be contours by:
```

Comment:
This seems to work great so it's fine, but in the future, and for more complex shapes, considered using [`cv2.match_shapes()`](http://docs.opencv.org/3.1.0/d5/d45/tutorial_py_contours_more_functions.html) which uses the Hu moment Invariants and should be much more robust than ad-hoc matching heuristics you come up with.